include/soci/soci-platform.h

```diff
@@ -8,6 +8,26 @@
 #ifndef SOCI_PLATFORM_H_INCLUDED
 #define SOCI_PLATFORM_H_INCLUDED
 
+// Check some C++11 Features
+#if     defined(_MSC_VER)
+// Bug in Visual Studio: __cplusplus still means C++03
+// https://connect.microsoft.com/VisualStudio/feedback/details/763051/
+#define SOCI_HAVE_NOEXCEPT (_MSC_VER >= 1900)
+#define SOCI_HAVE_LAMBDA   (_MSC_VER >= 1600)
+#elif   defined(__clang__) || defined(__GNUG__)
```

Comment:
Line 65 in SociConfig.cmake throws a FATAL_ERROR when using a compiler other than clang/gcc/vc. That is reflected by this change.
And I didn't even know that Oracle has a C++ compiler. Neither do I know if __cplusplus is defined correctly for compilers other than MSVC,gcc and clang nor do I know if CMake reports this compiler as being something else nor if all other compilers have implemented noexcept.
That's why I said I was trying to be conservative.
Unless you know these compilers well and you are about to remove this line 65 in SociConfig.cmake I'd keep the change like I proposed and leave the necessary changes to a user of other compilers.
When you simply refer to __cplusplus when not using MSVC you might force users of untested other compilers exactly into the work I had to exercise: Why is the simple test program abort()ing right away. There is no message like "you shouldn't throw in a destructor". The message reads "abort() called".
When you error out for unknown compilers then you're pointing their users directly to the important stuff to consider.
I'd favor the latter annoyance over than tracing through the test code.